src/components/context/SettingsContext.tsx

```diff
@@ -0,0 +1,113 @@
+import _ from 'lodash';
+
+import React, {
+  FC,
+  ReactElement,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { AppSetting } from '@graasp/apps-query-client';
+
+import { GENERAL_SETTINGS_KEY } from '../../config/appSettingsTypes';
+import { MUTATION_KEYS, hooks, useMutation } from '../../config/queryClient';
+import { DEFAULT_GENERAL_SETTINGS } from '../../config/settings';
+import { GeneralSettings } from '../../interfaces/settings';
+import Loader from '../common/Loader';
+
+export type SettingsContextType = {
+  settings: GeneralSettings;
+  changeSetting: (name: string, value: unknown) => void;
+  saveSettings: () => void;
+  resetSettings: () => void;
+  unsavedChanges: boolean;
+};
+
+const SettingsContext = createContext<SettingsContextType>({
+  settings: DEFAULT_GENERAL_SETTINGS,
+  changeSetting: () => null,
+  saveSettings: () => null,
+  resetSettings: () => null,
+  unsavedChanges: false,
+});
+
+type Prop = {
+  children: ReactElement | ReactElement[];
+};
+
+export const SettingsProvider: FC<Prop> = ({ children }) => {
+  const postSettings = useMutation<unknown, unknown, Partial<AppSetting>>(
```

Comment:
Probably if we have to define everytime the types, we might create custom hooks. It might be nice also so we don't need to import useMutation and the mutation keys. It can be a future improvement.
```
postAppSetting = useMutation<...>(KEY)
```